src/shared/clients/defaultS3Client.ts

```diff
@@ -62,7 +62,10 @@ export class DefaultS3Client implements S3Client {
             await s3
                 .createBucket({
                     Bucket: request.bucketName,
-                    CreateBucketConfiguration: { LocationConstraint: this.regionCode },
+                    // Passing us-east-1 for LocationConstraint breaks creating bucket. To make a bucket in us-east-1, you need to
+                    // not pass a region, so check for this case.
```

Comment:
it is very useful to link to code where possible:

```suggestion
                    // not pass a region, so check for this case.
                    // https://github.com/aws/aws-sdk-java-v2/blob/82aa860b0b29e00042a7d46e456e8ba2097b5f54/services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/handlers/CreateBucketInterceptor.java#L52
```

Reply:
It's not actually a thing from the java sdk v2 though (I actually got it from a boto3 issue), it's just the logic of the api.

Reply:
whatever reference explains this special case is worth linking to